### ISSUE TITLE
Question - Climate Control for Stream Deck Plus

### ISSUE THREAD

**Issue by kramttocs** [cgiesche/streamdeck-homeassistant#198]
Very excited to see Stream Deck Plus support.

Has anyone worked on setting up controlling a climate entity via a dial?
Not sure how to get started with making the dial execute the set target temperature service with [Current Temp] + or - [ticks]

Is it also possible to change the background dynamically based on the hvac mode? Flame, snowflake, etc

**Comment by jer78** [comment]
I posted something today that may help. It's using the dial to +/- for volume on a media_player. You could modify it for use with a climate device. The tricky part would be to limit it so it doesn't turn the temp to any extreme. Perhaps, have the automation first update a sensor and a few seconds delay before it actually sends to your thermostat. This would allow you to see what temperature it will change to before it sends that to your thermostat. If you do this, make sure the automation mode is set to "restart". 

https://github.com/cgiesche/streamdeck-homeassistant/issues/188#issuecomment-1836712121

**Comment by kramttocs** [comment]
Thank you! I'll look over that. Good call on the runaway part.  I wonder if logic could be added in the stream deck template where if the new value is > 75 (for example) it could just send 75. So keep the limit localized. Fortunately as a backup, most thermostats have a max and min threshold.

**Comment by kramttocs** [comment]
Yeah, I agree it'd be easier handling this in HA but I am not ready to write off doing it in the button logic yet :) Looking at the nunjucks referenced doc it allows for a decent amount of logic.
But...
I need to wrap my head around some of the more basic level templating/features of this plugin first and I appreciate the offer.

**Comment by cgiesche** [comment]
Hi kramttocs,

it is not possible to change the background image of the display via templating (in the plugin). But the plugin tries to use the icon that is provided by home assitant in the state attribute "icon". I am not a home-assistant expert, but maybe this may be a way to get state-dependent icons.
